src/Bundle/Maker/Factory/FactoryGenerator.php

```diff
@@ -130,19 +126,27 @@ private function createMakeFactoryData(Generator $generator, string $class, Make
             if (!$metadata || $metadata instanceof ODMClassMetadata && $metadata->isEmbeddedDocument) {
                 $persisted = false;
             }
+
+            if ($persisted) {
+                $repository = new \ReflectionClass($this->managerRegistry->getRepository($object->getName()));
+            }
         }
 
         return new MakeFactoryData(
             $object,
             $factory,
             $repository ?? null,
-            $this->phpstanEnabled(),
+            $this->staticAnalysisTool(),
             $persisted
         );
     }
 
-    private function phpstanEnabled(): bool
+    private function staticAnalysisTool(): string
     {
-        return \file_exists("{$this->kernel->getProjectDir()}/vendor/phpstan/phpstan/phpstan");
+        return match(true) {
+            \file_exists($this->kernel->getProjectDir().self::PHPSTAN_PATH) => MakeFactoryData::STATIC_ANALYSIS_TOOL_PHPSTAN,
+            \file_exists($this->kernel->getProjectDir().self::PSALM_PATH) => MakeFactoryData::STATIC_ANALYSIS_TOOL_PSALM,
+            default => MakeFactoryData::STATIC_ANALYSIS_TOOL_NONE,
+        };
```

Comment:
this does not covers the case where the user uses BOTH PHPStan and Psalm (I know this exists)

but this will add a so long phpdoc, that I didn't wanted to do that!